blobstore/list.go

```diff
@@ -27,6 +27,31 @@ type ListResult struct {
 	ModifiedBy string    `json:"modified_by"`
 }
 
+// CheckAndAdjustPrefix checks if the prefix is an object and adjusts the prefix accordingly.
+// Returns the adjusted prefix, an error message (if any), and the HTTP status code.
+func CheckAndAdjustPrefix(s3Ctrl *S3Controller, bucket, prefix string) (string, string, int) {
```

Comment:
I think it would be more standard to return an error, instead of a errorMessage? Like (string, int, error)